fence/fence.proto

```diff
@@ -0,0 +1,86 @@
+// Code generated by make; DO NOT EDIT.
+syntax = "proto3";
+package fence;
+
+import "google/protobuf/descriptor.proto";
+
+option go_package = ".;fence";
+
+extend google.protobuf.FieldOptions {
+  // Indicates that a field MAY contain information that is sensitive
+  // and MUST be treated as such (e.g. not logged).
+  bool fence_secret = 1099;
+}
+
+// Controller holds the RPC method for performing fencing operations.
+service Controller {
+  // EnableClusterFence RPC call to perform a fencing operation.
+  rpc EnableClusterFence (EnableClusterFenceRequest)
+  returns (EnableClusterFenceResponse) {}
+
+  // DisableClusterFence RPC call to remove a list of CIDR blocks from the
+  // list of blocklisted/fenced clients.
+  rpc DisableClusterFence (DisableClusterFenceRequest)
+  returns (DisableClusterFenceResponse) {}
+
+  // ListClusterFence RPC call to provide a list of blocklisted/fenced clients.
+  rpc ListClusterFence(ListClusterFenceRequest)
+  returns (ListClusterFenceResponse){}
+}
+// EnableClusterFenceRequest contains the information needed to identify
+// the cluster so that the appropriate fencing operation can be
+// performed.
+message EnableClusterFenceRequest {
+  // Plugin specific parameters passed in as opaque key-value pairs.
+  map<string, string> parameters = 1;
+  // Secrets required by the plugin to complete the request.
+  map<string, string> secrets = 2 [(fence_secret) = true];
+  // list of CIDR blocks on which the fencing operation is expected to be
+  // performed.
+  repeated CIDRList cidr_list = 3;
+}
+
+// EnableClusterFenceResponse holds the information about the result of the
+// EnableClusterFenceRequest call.
+message EnableClusterFenceResponse {
+    // The message returned on execution as received from SP.
+    string result = 1;
```

Comment:
I had suggested for the addition of this response field. However, after checking with @raghavendrabhat, I am not sure. @Madhu-1 what is your opinion? Is there any benefit to passing the string response of the specific storage implementation back or would a boolean suffice?

Reply:
I think it will not be the best approach to just return the unprocessed output string; as the output of a successful blocklist is somewhat similar to :
```blocklisting 10.244.0.64:0/0 until 2021-12-02T03:12:32.490358+0000 (60 sec)``` .

Here: even if this is received from the output, the gRPC caller already knows which IPs are getting blocklisted, as it is the one sending request for the same in the first place.  
The only thing the gRPC caller should know is whether the call was successful or not(bool).
Wdyt? @Madhu-1 @nixpanic

Reply:
@raghavendra-talur what is the reason to get a `result`? It would be good to explain for what it will be used.

gRPC returns a `(message, error)` on requests, so if there were issues with doing the fence operation, I expect to see suitable errors returned (with their own error-string).

Reply:
we don't need any message in case of success as it's not much helpful we can have empty  `EnableClusterFenceResponse{}` and error